helix-core/src/syntax.rs

```diff
@@ -1606,11 +1771,23 @@ impl HighlightConfiguration {
     }
 }
 
-impl<'a> HighlightIterLayer<'a> {
-    // First, sort scope boundaries by their byte offset in the document. At a
-    // given position, emit scope endings before scope beginnings. Finally, emit
-    // scope boundaries from deeper layers first.
-    fn sort_key(&mut self) -> Option<(usize, bool, isize)> {
+trait IterLayer {
+    // TODO: this can be cleaner with GATs once the MSRV is 1.65:
+    // type SortKey where Self: PartialOrd;
+    type SortKey;
```

Comment:
I don't udnerstand what GATs woul be used for here.
`type SortKey: PartialOrd` seems to be sufficient here.

Reply:
Oh hah actually that's just me misunderstanding traits and GATs 😄. This works great